presentation/src/main/java/me/androidbox/presentation/login/screen/LoginScreen.kt

```diff
@@ -17,22 +20,46 @@ import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
-import androidx.hilt.navigation.compose.hiltViewModel
 import me.androidbox.component.R
 import me.androidbox.component.general.PasswordTextField
 import me.androidbox.component.general.TaskButton
 import me.androidbox.component.general.UserInputTextField
 import me.androidbox.component.ui.theme.BusbyTaskyTheme
 import me.androidbox.component.ui.theme.backgroundInputEntry
-import me.androidbox.presentation.login.viewmodel.LoginViewModel
+import me.androidbox.domain.authentication.ResponseState
+import me.androidbox.domain.authentication.model.Login
 
 @Composable
 fun LoginScreen(
     modifier: Modifier = Modifier,
-    loginViewModel: LoginViewModel = hiltViewModel(),
-    onSignUpClicked: () -> Unit
+    loginState: State<ResponseState<Login>?>,
+    onLoginSuccess: (login: Login) -> Unit,
+    onSignUpClicked: () -> Unit,
+    email: String,
+    onEmailChanged: (email: String) -> Unit,
+    password: String,
+    onPasswordChanged: (newPassword: String) -> Unit,
+    onPasswordVisibilityChanged: () -> Unit,
+    isPasswordVisible: Boolean,
+    onLoginUser: (email: String, password: String) -> Unit
```

Comment:
To avoid having a ton of lambdas, you can have an "onEvent" method in your viewmodel, and send LoginEvents

```
sealed class LoginEvent {
    data class PasswordChange(val password: String) : LoginEvent()
    data class EmailChange(val email: String) : LoginEvent()
    object Login : LoginEvent()
// etc...
}
```

and on your viewmodel:
```
 fun onEvent(event: LoginEvent) {
        when (event) {
            is LoginEvent.EmailChange -> {
                state = state.copy(
                    email = event.email
                )
            }
            is LoginEvent.PasswordChange -> {
                state = state.copy(
                    password = event.password
                )
            }
// ETC...
        }
}
```

Reply:
And regarding passing the loginState: State<ResponseState<Login>?>, email, password, and so on

You could have just 1 state, that has every state related to the screen
Ex:

``` 
data class LoginState(
    val email: String = "",
    val emailError: String? = null,
    val password: String = "",
    val passwordError: String? = null,
    val isLoggedIn: Boolean = false,
    val isLoading: Boolean = false
// etc...
)
```